File: frontend/src/types/csv-import.ts
/**
 * CSV import types and interfaces
 */

export interface CSVColumnMapping {
  date?: string;
  amount?: string;
  description?: string;
  merchant?: string;
}

export interface CSVPreviewRow {
  date: string | null;
  amount: number | null;
  description: string;
  merchant: string;
  raw: Record<string, string>;
}

export interface CSVPreviewResponse {
  headers: string[];
  detected_mapping: CSVColumnMapping;
  preview_rows: CSVPreviewRow[];
  total_rows: number;
}

export interface CSVImportRequest {
  account_id: string;
  column_mapping: CSVColumnMapping;
  skip_duplicates?: boolean;
}

export interface CSVImportResponse {
  imported: number;
  skipped: number;
  errors: string[];
  total_processed: number;
}

File: backend/app/schemas/csv_import.py
"""CSV import schemas."""

from typing import Dict, List, Any, Optional
from uuid import UUID

from pydantic import BaseModel


class CSVPreviewRequest(BaseModel):
    """Request schema for CSV preview."""

    column_mapping: Optional[Dict[str, str]] = None


class CSVPreviewResponse(BaseModel):
    """Response schema for CSV preview."""

    headers: List[str]
    detected_mapping: Dict[str, Optional[str]]
    preview_rows: List[Dict[str, Any]]
    total_rows: int


class CSVImportRequest(BaseModel):
    """Request schema for CSV import."""

    account_id: UUID
    column_mapping: Dict[str, str]
    skip_duplicates: bool = True


class CSVImportResponse(BaseModel):
    """Response schema for CSV import."""

    imported: int
    skipped: int
    errors: List[str]
    total_processed: int
